### PR TITLE
Upgrade CI environment to Ubuntu 19.10 (eoan)

### PR DESCRIPTION
**Pull request by dnnr**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,54 +13,54 @@ branches:
...
         - name: "Build with most recent GCC & Clang, run tests"
-          env: TRAVIS_CACHE_ID=disco
+          env: TRAVIS_CACHE_ID=eoan
           cache:
               directories:
-                  - $HOME/.ccaches-disco
+                  - $HOME/.ccaches-eoan
                   - $HOME/.tox-cache
           script:
               # Fetch our current Ubuntu 19.04 CI image from Docker Hub
```
Comment (blarz) [inline .travis.yml:22]: Update the comment

File: .travis.yml [hunk 2]
```diff
@@ -13,54 +13,54 @@ branches:
...
               - bash <(curl -s https://codecov.io/bash) -y .codecov.yml -f coverage.info
 
         - name: "Run linters and static analyzers"
-          env: TRAVIS_CACHE_ID=disco-linter
+          env: TRAVIS_CACHE_ID=eoan-linter
           cache:
               directories:
                   - $HOME/.tox-cache
           script:
               # Fetch our current Ubuntu 19.04 CI image from Docker Hub
```
Comment (blarz) [inline .travis.yml:49]: Update the comment

File: .travis.yml (hunk 1 above)
Comment (dnnr) [reply .travis.yml]: Thanks!

File: .travis.yml [hunk 3]
```diff
@@ -13,55 +13,55 @@ branches:
...
 
               # Perform actual tasks within temporary Docker image
               - docker run -u $UID --privileged -t --rm
                 --volume=/etc/passwd:/etc/passwd:ro
                 --volume=$PWD:/hlwm:rw
                 --volume=$HOME/.tox-cache:/hlwm/.tox:rw
-                disco sh -c '
-                    /hlwm/ci/build.py --build-dir=/hlwm/build-clang-7 --cxx=clang++-7 --cc=clang-7 --build-type=Debug --check-using-std --iwyu --flake8
+                eoan sh -c '
+                    /hlwm/ci/build.py --build-dir=/hlwm/build-clang-9 --cxx=clang++-7 --cc=clang-9 --build-type=Debug --check-using-std --iwyu --flake8
```
Comment (blarz) [inline .travis.yml:64]: clang++-9

File: ci/Dockerfile.ci-eoan [hunk 4]
```diff
@@ -1,13 +1,13 @@
-FROM ubuntu:disco-20190204
+FROM ubuntu:eoan-20200313
 
 # Build deps
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install \
     ccache \
-    clang-7 \
+    clang-9 \
     clang-tidy-8 \
```
Comment (blarz) [inline ci/Dockerfile.ci-eoan:7]: Should we update clang-tidy here, too?

Comment (dnnr) [reply ci/Dockerfile.ci-eoan:7]: I actually wanted to do the tool upgrades later. Switching to the new compilers was not avoidable, though. In the case of clang-tidy, there seem to be some new issues with version 9, so I'd prefer doing the upgrade separately.

**Pull request merged**